src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1480,6 +1478,7 @@ static UCS_CLASS_CLEANUP_FUNC(uct_dc_mlx5_iface_t)
             &self->super.super.super.super.worker->super,
             &self->tx.dci_release_prog_id);
     uct_dc_mlx5_iface_dcis_destroy(self, uct_dc_mlx5_iface_total_ndci(self));
+    ucs_free(self->tx.dcis);
```

Comment:
should be done after 1486

Reply:
also better move ucs_arbiter_cleanup(&self->tx.dci_pool[pool_index].arbiter); from  UCS_CLASS_CLEANUP_FUNC(uct_dc_mlx5_iface_t) to uct_dc_mlx5_iface_dcis_destroy.
then unite the dci pools release+ucs_free(self->tx.dcis); to one helper function , used by both UCS_CLASS_INIT_FUNC (err flow) and UCS_CLASS_CLEANUP_FUNC

Reply:
fixed in another way,  it helps to simplify the code even more.
also, the 2nd commit fixes other memory leaks